Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/comments/CommentsDialog.java

```diff
@@ -269,9 +269,13 @@ private JPanel createPanel() {
 			JTextArea currentTextArea = getSelectedTextArea();
 			AnnotationAdapterWrapper aaw =
 				(AnnotationAdapterWrapper) annotationsComboBox.getSelectedItem();
-			currentTextArea.insert(aaw.getPrototypeString(),
-				currentTextArea.getCaretPosition());
-			currentTextArea.setCaretPosition(currentTextArea.getCaretPosition() - 1);
+			if (currentTextArea.getSelectedText() != null && !currentTextArea.getSelectedText().isEmpty()) {
```

Comment:
There is code below performing effectively the same function as your new code (starting at line 362 in the original code).   Both code blocks should be refactored into one method that can be called from both places.   The code will look something like this (before my other change requests have been applied):

```
	private void insertAnnotation(JTextArea textArea, AnnotationAdapterWrapper annotation) {
		textArea.insert(annotation.getPrototypeString(), textArea.getCaretPosition());
		textArea.setCaretPosition(textArea.getCaretPosition() - 1);
	}
```